test/intl402/DateTimeFormat/prototype/formatToParts/related-year-zh.js

```diff
@@ -12,18 +13,25 @@ locale: [zh-u-ca-chinese]
 function verifyFormatParts(actual, expected, message) {
   assert.sameValue(Array.isArray(expected), true, `${message}: expected is Array`);
   assert.sameValue(Array.isArray(actual), true, `${message}: actual is Array`);
-  assert.sameValue(actual.length, expected.length, `${message}: length`);
+  if (actual.length !== expected.length)
+    return false;
 
   for (let i = 0; i < actual.length; ++i) {
     assert.sameValue(actual[i].type, expected[i].type, `${message}: parts[${i}].type`);
     assert.sameValue(actual[i].value, expected[i].value, `${message}: parts[${i}].value`);
   }
+  return true;
 }
 
 const df = new Intl.DateTimeFormat("zh-u-ca-chinese", {year: "numeric"});
 const date = new Date(2019, 5, 1);
-verifyFormatParts(df.formatToParts(date), [
+const expected1 = [
   {type: "relatedYear", value: "2019"},
   {type: "yearName", value: "己亥"},
   {type: "literal", value: "年"},
-]);
+];
+const expected2 = [
+  {type: "yearName", value: "己亥"},
+  {type: "literal", value: "年"},
+];
+assert.sameValue(verifyFormatParts(df.formatToParts(date), expected1) || verifyFormatParts(df.formatToParts(date), expected2), true);
```

Comment:
We can remove the whole `verifyFormatParts` function, and then:

```suggestion
const actual = df.formatToParts(date);

assert.sameValue(Array.isArray(actual), true, 'actual is Array');

if (actual.length > 2) {
  expected.shift(); // removes the relatedYear
}

actual.forEach(({ type, value }, i) => {
  const { type: eType, value: eValue } = expected[i];
  assert.sameValue(type, eType, `actual[${i}].type should be ${eType}`);
  assert.sameValue(value, eValue, `actual[${i}].value should be ${eValue}`);
}); 
```

Reply:
Sounds pretty nice. Fixing.

Reply:
if you prefer (but not limited to) you can apply these changes through the GitHub interface. I should probably squash them at the end.

Reply:
I've updated the PR by squashing changes :)